src/Query.tsx

```diff
@@ -320,28 +305,32 @@ export default class Query<TData = any, TVariables = OperationVariables> extends
   private resubscribeToQuery() {
     this.removeQuerySubscription();
 
-    const lastError = this.queryObservable!.getLastError();
-    const lastResult = this.queryObservable!.getLastResult();
+    const queryObservable = this.state.queryObservable!;
+    // probably need to move this to state
+    const lastError = queryObservable.getLastError();
+    const lastResult = queryObservable!.getLastResult();
     // If lastError is set, the observable will immediately
     // send it, causing the stream to terminate on initialization.
     // We clear everything here and restore it afterward to
     // make sure the new subscription sticks.
-    this.queryObservable!.resetLastResults();
-    this.startQuerySubscription();
-    Object.assign(this.queryObservable!, { lastError, lastResult });
+    queryObservable!.resetLastResults();
+    this.startQuerySubscription(queryObservable);
+    Object.assign(queryObservable!, { lastError, lastResult });
   }
 
   private updateCurrentData = () => {
-    // force a rerender that goes through shouldComponentUpdate
     if (this.hasMounted) this.forceUpdate();
```

Comment:
I'm not particularly excited about throwing promises, but I _am_ excited about the day where you can change this line to do something like this:

```js
if (this.hasMounted) {
  if (this.props.priority === 'low') {
    ReactDOM.unstable_deferredUpdate(() => this.forceUpdate());
  } else {
    this.forceUpdate();
  }
}
```

Naturally, I also hope that the React team decouples this scheduling primitive from `ReactDOM` and provides further granularity (e.g., explicit scheduling timeouts), but hey, this is _the_ value prop for me.